drivers/platform/nintendo3ds/ctr_gpiointc.c

```diff
@@ -21,50 +21,26 @@
 #include <linux/mod_devicetable.h>
 #include <linux/irqchip/chained_irq.h>
 
-#define to_gpiointc(map)	\
-	container_of(map, struct ctr_gpiointc, map[(map)->lid])
-
-#define REGOFF_TYPE(intc, i)	(i)
-#define REGOFF_ENABLE(intc, i)	((intc)->nreg + (i))
-
-struct ctr_gpiointc_map {
-	int lid;
-	int hwid;
-
-	int virq;
-	struct gpio_desc *gpio;
-};
-
 struct ctr_gpiointc {
 	struct device *dev;
 
-	int nreg;
-	void __iomem *regs;
-
 	int irq_base;
 	struct irq_domain *irqdom;
 	struct irq_chip_generic *irqgc;
 
-	int nr_irqs;
-	struct ctr_gpiointc_map map[];
+	int virq;
+	struct gpio_desc *in_gpio;
```

Comment:
does this need to be a member of `struct ctr_gpiointc` or can it be local to `ctr_gpiointc_domap`?  Should it be set somewhere?

Reply:
they can be local to the mapping function, all I need is to `get` the gpio in INPUT mode